impl/observable-polyfill.d.ts

```diff
@@ -0,0 +1,379 @@
+import { Observable } from './observable';
+
+declare global {
+	interface EventTarget {
+		on(eventType: string): Observable<Event>;
+	}
+	interface AbortSignal {
+		on<K extends keyof AbortSignalEventMap>(
+			eventType: K,
+		): Observable<AbortSignalEventMap[K]>;
+	}
+	interface AbstractWorker {
+		on<K extends keyof AbstractWorkerEventMap>(
+			eventType: K,
+		): Observable<AbstractWorkerEventMap[K]>;
+	}
+	interface Animation {
```

Comment:
Why are all of these being added?

Reply:
Honestly? Just to be nice. If anyone wants to use this with TypeScript, these types will help them. They don't *need* to be here.

Reply:
My two cents is we leave them. They can't do anything but help folks use this and try it out.

Reply:
They map the event name strings to the event subclasses. This way `eventTarget.on('click')` will typed as an `Observable<MouseEvent>` while `eventTarget.on('keypress')` will be typed as an `Observable<KeyboardEvent>`.